kernel/dma/page_touching.c

```diff
@@ -132,3 +133,23 @@ void setup_dma_page_touching_ops(struct device *dev)
 	dev_info(dev, "binding to page touching DMA ops\n");
 	dev->dma_ops = &page_touching_dma_ops;
 }
+
+static const struct dmi_system_id pt_enable_table[] __initconst = {
+	{
+		.matches = {
+			DMI_EXACT_MATCH(DMI_SYS_VENDOR, "Amazon EC2"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "caspian"),
```

Comment:
How does this product name match work? Should the instance type have caspian in its name?

Reply:
I thought product_name was the instance type; something more specific like caspianr1g.8xlarge.

How can I verify this product name my side - is it in dmidecode?

Reply:
Yes, that's what it's checking for.

Reply:
`sudo dmesg | grep DMI`

Reply:
Ah, I see that DMI_MATCH seems to look for a substring. So, yeah, LGTM.

```
  8 /*
  7  * struct dmi_device_id appears during expansion of
  6  * "MODULE_DEVICE_TABLE(dmi, x)". Compiler doesn't look inside it
  5  * but this is enough for gcc 3.4.6 to error out:
  4  *      error: storage size of '__mod_dmi_device_table' isn't known
  3  */
  2 #define dmi_device_id dmi_system_id
  1 
601 #define DMI_MATCH(a, b) { .slot = a, .substr = b }
  1 #define DMI_EXACT_MATCH(a, b)   { .slot = a, .substr = b, .exact_match = 1 }
  2 
```

Amazon Linux maintainers - please can this be applied to the patch file for all AL versions? I think this driver should exist in all AL.

Reply:
Testing:
```
 [root@ip-10-0-174-244 ec2-user]# dmesg | grep touching
 [    0.000000] Kernel command line: BOOT_IMAGE=(hd0,gpt1)/boot/vmlinuz-6.1.99+ root=UUID=76d95e62-fab5-42eb-be84-bb6a087a2b13 ro console=tty0 console=ttyS0,115200n8 nvme_core.io_timeout=4294967295 rd.emergency=poweroff rd.shell=0 selinux=1 security=selinux numa_cma=1:64M page_touching.dma_page_touching_enable=y panic=1
 [    0.061954] Automatically enabling page touching for Caspian
 [    0.118379] serial 0000:00:01.0: binding to page touching DMA ops
 [   16.226484] nvme 0000:00:04.0: binding to page touching DMA ops
 [   24.864414] virtio-pci 0000:00:1f.0: binding to page touching DMA ops
 [   33.587934] ena 0000:00:05.0: binding to page touching DMA ops
```

Reply:
Hm, that's my initial testing version where I *also* added it on the command line just in case. Here's it really *working*:

```
[ec2-user@ip-10-0-174-244 ~]$ sudo dmesg | grep touching
[    0.061686] Automatically enabling page touching for Caspian
[    0.106705] serial 0000:00:01.0: binding to page touching DMA ops
[   16.202393] nvme 0000:00:04.0: binding to page touching DMA ops
[   24.896716] virtio-pci 0000:00:1f.0: binding to page touching DMA ops
[   33.751150] ena 0000:00:05.0: binding to page touching DMA ops
```

Reply:
Sounds good, thanks

Reply:
@dwmw2 We have merged this pull request to our AL2 5.10 & AL2023 6.1 kernels and it should be part of our upcoming kernel release.